traits/has_traits.py

```diff
@@ -2188,6 +2188,20 @@ def class_editable_traits ( cls ):
         return names
 
     class_editable_traits = classmethod( class_editable_traits )
+    
+    def visible_traits ( self ):
+        """Returns an alphabetically sorted list of the names of non-event
+        trait attributes associated with the current object, that should be GUI visible
+        """
+        return self.trait_names( type = not_event, editable = not_false, visible = not_false )
```

Comment:
Changed behavior accordingly. I agree that it's cleaner like this, however the default changed (to show non-editable traits) and therefore if there are internal traits that shall not be shown they must set the visible=False property.